crates/wasi-http/src/struct.rs

```diff
@@ -89,17 +132,26 @@ impl From<Bytes> for Stream {
     }
 }
 
+impl From<&mut Stream> for bytes::Bytes {
+    fn from(stream: &mut Stream) -> Self {
+        stream.data.clone().into()
```

Comment:
I don't like when clones are hidden inside unassuming impls like From, and `From<&mut _>` is unidiomatic as well. `impl Stream { fn as_bytes(&self) -> Self { Bytes::from(self.data.clone()) }}` would be a better way to write this one